### PR TITLE
Accessibility improvements

### PR DESCRIPTION
**Pull request by mullermp**
# Description

For AWS SDK for Ruby, we had to make some accessibility fixes to the docs, involving some styling and color changing. I think some of these may be helpful to contribute back.

I have not written tests yet because I want to know if these are changes that we can (and should) make before I go and invest that time.

# Completed Tasks

- [x] I have read the [Contributing Guide][contrib].
- [ ] The pull request is complete (implemented / written).
- [x] Git commits have been cleaned up (squash WIP / revert commits).
- [ ] I wrote tests and ran `bundle exec rake` locally (if code is attached to PR).

[contrib]: https://github.com/lsegal/yard/blob/main/CONTRIBUTING.md

### CODE REVIEW

File: templates/default/fulldoc/html/css/style.css [hunk 1]
```diff
@@ -82,6 +82,12 @@ body {
   #search { display: none; }
 }
 
+@media (max-width: 320px) {
```
Comment (mullermp) [inline templates/default/fulldoc/html/css/style.css:85]: We had a requirement to support devices with < 320 px width (like smart phone). I'm not a CSS expert, but the way this is coded in style.css for 920px min-width interferes with 320px unless I copy over the same properties. I effectively am only adding break-word here. I am probably not the best person to untangle / cascade this properly. It removes all horizontal scrolling except for tables and code snippets which is acceptable for the requirement.

File: templates/default/fulldoc/html/css/style.css [hunk 2]
```diff
@@ -106,6 +112,7 @@ h2 small a {
   position: relative;
   padding: 2px 7px;
 }
+a { font-weight: 550; }
```
Comment (mullermp) [inline templates/default/fulldoc/html/css/style.css:115]: Bold links are 700, normal is 400. This is a weight in between. The requirement was to make links more obvious for people with low vision.

File: templates/default/fulldoc/html/full_list.erb [hunk 3]
```diff
@@ -26,7 +26,10 @@
           <% end %>
         </div>
 
-        <div id="search">Search: <input type="text" /></div>
+        <div id="search">
+          <label for="search-class">Search:</label>
+          <input id="search-class" type="text" />
+        </div>
```
Comment (mullermp) [inline templates/default/fulldoc/html/full_list.erb:32]: This is just adding a label for search for screen readers to know what it's for.

File: templates/default/fulldoc/html/full_list.erb [hunk 4]
```diff
@@ -1,5 +1,5 @@
 <!DOCTYPE html>
-<html>
+<html <%= "lang=\"#{html_lang}\"" unless html_lang.nil? %>>
```
Comment (mullermp) [inline templates/default/fulldoc/html/full_list.erb:2]: We hardcoded lang="en" in our docs, but I think it makes sense to have this as a setup method.

File: templates/default/tags/html/example.erb [hunk 5]
```diff
@@ -1,9 +1,9 @@
 <% if object.has_tag?(:example) %>
   <div class="examples">
-    <p class="tag_title">Examples:</p>
+    <h4 class="tag_title">Examples:</h4>
```
Comment (mullermp) [inline templates/default/tags/html/example.erb:3]: One requirement we had was to not have paragraph sections for examples. We set these to h4 and h5 headers so I think this is reasonable as they are supposed to be sections.

File: templates/default/fulldoc/html/setup.rb [hunk 6]
```diff
@@ -225,15 +231,17 @@ def class_list(root = Registry.root, tree = TreeContext.new)
     has_children = run_verifier(child.children).any? {|o| o.is_a?(CodeObjects::NamespaceObject) }
     out << "<li id='object_#{child.path}' class='#{tree.classes.join(' ')}'>"
     out << "<div class='item' style='padding-left:#{tree.indent}'>"
-    out << "<a class='toggle'></a> " if has_children
+    accessible_props = "aria-label='#{name} child nodes' aria-expanded='false' aria-controls='object_#{child.path}'"
```
Comment (mullermp) [inline templates/default/fulldoc/html/setup.rb:234]: Setting standard aria properties. The JS code will flip these.

File: templates/default/fulldoc/html/js/full_list.js [hunk 7]
```diff
@@ -154,10 +171,12 @@ function partialSearch(searchString, offset) {
...
   searchTimeout = null;
   highlight();
-  if ($('#full_list li:visible').size() === 0) {
+  var found = $('#full_list li:visible').size();
+  if (found === 0) {
     $('#noresults').text('No results were found.').hide().fadeIn();
   } else {
-    $('#noresults').text('').hide();
+    // This is read out to screen readers
+    $('#noresults').text('There are ' + found + ' results.').hide().fadeIn();
```
Comment (mullermp) [inline templates/default/fulldoc/html/js/full_list.js:179]: I wanted to change noresults to searchresults or something, but I figure I cannot change this for existing templates out there. I've just updated it to output the results at the bottom of the list, and it's read out to screen readers.

File: templates/default/fulldoc/html/js/full_list.js [hunk 8]
```diff
@@ -91,7 +108,7 @@ function enableSearch() {
     }
   });
 
-  $('#full_list').after("<div id='noresults' style='display:none'></div>");
+  $('#full_list').after("<div id='noresults' role='status' style='display:none'></div>");
```
Comment (mullermp) [inline templates/default/fulldoc/html/js/full_list.js:111]: Role is added for screen reader.

File: templates/default/fulldoc/html/js/full_list.js [hunk 9]
```diff
@@ -62,8 +62,25 @@ function enableToggles() {
...
     evt.preventDefault();
     $(this).parent().parent().toggleClass('collapsed');
+    $(this).attr('aria-expanded', function (i, attr) {
+        return attr == 'true' ? 'false' : 'true'
+    });
     highlight();
   });
+
+  // navigation of nested classes using keyboard
+  $('#full_list a.toggle').on('keypress',function(evt) {
```
Comment (mullermp) [inline templates/default/fulldoc/html/js/full_list.js:72]: Part of this change allows you to tab navigate to the arrow/carrot in the full list menu, then you can press enter to expand the section without visiting a new page.

**Review by mullermp — COMMENTED** [review]
Just adding some comments to explain code changes.

**Comment by lsegal** [comment]
Sorry for the delay but thank you for the contribution!

**Comment by mullermp** [comment]
Let me double check this week that these are the latest ones I've made. There may be some more.

**Comment by mullermp** [comment]
@lsegal

Here are the overrides we eventually went with: https://github.com/aws/aws-sdk-ruby/blob/version-3/doc-src/templates/default/fulldoc/html/css/common.css

Looks like there is a minor difference here - I committed
```
body { height: 100%; overflow-wrap: break-word; }
```
but in PR there's 
```
body { height: 100%; overflow: hidden; overflow-wrap: break-word; }
```

but also, I'm not a styling guru, please double check everything :)

**Pull request merged**